pegen/grammar.py

```diff
@@ -386,192 +386,22 @@ def initial_names(self) -> AbstractSet[str]:
         return self.rhs.initial_names()
 
 
+class Cut:
+
+    def __init__(self):
+        pass
+
+    def __repr__(self):
+        return f"Cut()"
+
+    def __str__(self):
+        return f"~"
+
+    def __eq__(self, other):
+        if not isinstance(other, Cut):
+            return NotImplemented
+        return True
+
+
 Plain = Union[Leaf, Group]
-Item = Union[Plain, Opt, Repeat, Lookahead, Rhs]
-
-
-class GrammarParser(Parser):
```

Comment:
In story6 I kept the old grammar parser around just so that I could still parse the meta-grammar even if I had screwed up something that broke the generated meta-parser. Check out the `-b` flag there.

Reply:
I can do this, but one thing to notice is that that with the new rules (especially the possibility to write every option in a new line) the old grammar parser cannot parse the metagrammar. This is why I have eliminated it completely.

Reply:
OK, that's fine. I think for story6 I updated the hand-written grammar parser, but since we use Git we don't really need a backup anyway... :-)